mlflow/sagemaker/__init__.py

```diff
@@ -1376,7 +1376,8 @@ def _truncate_name(name, max_length):
 
 
 def _get_unique_name(base_name, unique_suffix, unique_id_length=20):
-    unique_resource_string = f"{unique_suffix}{get_unique_resource_id(unique_id_length)}"
```

Comment:
Another option is to fix `get_unique_resource_id` to only return a string with `[0-9a-z]`.